tests/jax/test_jax_3d_eldry.py

```diff
@@ -0,0 +1,1005 @@
+        
+        obj.assertLess(dFH_error, df_tol, 
+                        'Incorrect Tangential AFT gradient in X direction.')
+        
+        
+        
+        FnlH_vec, dFnldUH_vec, dFnldw_vec \
+            = eldry2d.aft(Unl[tyn,:], w, h, Nt=Nt)
+        
+        FnlH, dFnldUH, dFnldw = obj.eldry_force.aft(Unl, w, h, Nt=Nt)
+        
+        
+        FH_error = np.max(np.abs(FnlH[tyn]-FnlH_vec))
+        
+        
+        obj.assertLess(FH_error, force_tol, 
+                        'Incorrect elastic dry friction force in Y direction.')
+        
+    
+        
+        ###############
+        # Tangent - Tangent Gradient in Y direction
+        dFH_error = np.max(np.abs(dFnldUH[np.ix_(tyn,tyn)]-dFnldUH_vec))
+        
+        obj.assertLess(dFH_error, df_tol, 
+                        'Incorrect Tangential AFT gradient in Y direction.')
+        ###############
+        # Numeric Gradient check, should capture dTangent/dNormal terms
+        
+        # Check gradient - Unl
+        fun = lambda U : obj.eldry_force.aft(U, w, h, Nt=Nt)[0:2]
+        
+        grad_failed = vutils.check_grad(fun, Unl, verbose=False, 
+                                        atol=obj.atol_grad, h=delta_grad)
+        
+        obj.assertFalse(grad_failed, 'Incorrect Gradient w.r.t. Unl.')
+        
+        # Gradient w
+        fun = lambda w : obj.eldry_force.aft(Unl, w, h, Nt=Nt)[0::2]
+        
+        grad_failed = vutils.check_grad(fun, np.array([w]), verbose=False, 
+                                        atol=obj.atol_grad)
+
+        obj.assertFalse(grad_failed, 'Incorrect Gradient w.r.t. w.')
+        
+class TestJAXEldry(unittest.TestCase):
+    
+    def __init__(self, *args, **kwargs):
+        """
+        Define tolerances here for all the tests
+
+        Returns
+        -------
+        None.
+
+        """
+        super(TestJAXEldry, self).__init__(*args, **kwargs)      
+        
+        
+        force_tol = 5e-15 # All should be exactly equal
+        df_tol = 1e-14 # rounding error on the derivatives
+        dfdw_tol = 1e-16 # everything should have zero derivative w.r.t. frequency
+        self.atol_grad = 1e-9
+
+        self.tols = (force_tol, df_tol, dfdw_tol)
+
+
+        # Simple Mapping to displacements - eldry
+        Q = np.array([[1.0, 0.0, 0.0], [0.0, 1.0, 0.0],[0.0, 0.0, 1.0]])
+        T = np.array([[1.0, 0.0, 0.0], [0.0, 1.0, 0.0],[0.0, 0.0, 1.0]])
+        
+        # Mapping for 2D elastic dry x direction
+        Q_2D = np.array([[1.0, 0.0], [ 0.0, 1.0]])
+        T_2D = np.array([[1.0, 0.0], [0.0, 1.0]])
+        
+
+        
+        kt = 2.0
+        kn = 2.5
+        mu = 0.75
+        
+        self.un_low = 1.5
+        self.un_high = 5.7
+        
+
+        
+        self.eldry2D_force = ElasticDryFriction2D(Q_2D, T_2D, kt, kn, mu, 
+                                              u0=np.array([0.0]))
+        
+        self.eldry_force = ElasticDryFriction3D(Q, T, kt, kn, mu, 
```

Comment:
Can you put `3D` in the name that way it is easy reading the variable names to know which are 2D and 3D? Otherwise, I see that some are 2D, but the variable names for the others are vague.

Likewise for other variables below.